lib/posix/fs.c

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <string.h>
+#include <errno.h>
+#include <posix/unistd.h>
+
+
+struct p_fs_file {
+	struct fs_file_t file;
+	bool used;
+};
+
+struct p_fs_dir {
+	struct fs_dir_t	dir;
+	bool used;
+};
+
+static struct p_fs_dir dir_array[CONFIG_MAX_OPEN_FILES];
+K_MUTEX_DEFINE(dirp_mutex);
+
+static struct p_fs_file file_array[CONFIG_MAX_OPEN_FILES];
+K_MUTEX_DEFINE(filep_mutex);
```

Comment:
Why a mutex?  Unless you really need priority inheritance (and AFAICT you don't), you don't want this.  Use either a semaphore (if you want blocking - again AFAICT you don't) or an irq_lock() (if you can get away with just a critical section).  Or if you want to be fancy use a spinlock, which becomes a non-recursive irq_lock() in uniprocessor mode and is more futureproof.